docs/tutorials/excel-tutorial-create-custom-functions.md

```diff
@@ -3,8 +3,293 @@ title: Excel custom functions tutorial
+
+1. Run the following command and then answer the prompts as follows.
+
+    ```
+    yo office
+    ```
+
+    * Choose a project type: `Excel Custom Functions Add-in project (...)`
+
+    * Choose a script type: `JavaScript`
+
+    * What do you want to name your add-in? `stock-ticker`
+
+    ![Yeoman generator for Office Add-ins prompts for custom functions](../images/12-10-fork-cf-pic.jpg)
+
+    The Yeoman generator will create the project files and install supporting Node components. The project files come from the [Excel-Custom-Functions](https://github.com/OfficeDev/Excel-Custom-Functions) GitHub repository.
+
+2. Go to the project folder.
+
+    ```
+    cd stock-ticker
+    ```
+
+3. Trust the self-signed certificate that is needed to run this project. For detailed instructions for either Windows or Mac, see [Adding Self Signed Certificates as Trusted Root Certificate](https://github.com/OfficeDev/generator-office/blob/master/src/docs/ssl.md).  
+
+4. Build the project.
+
+    ```
+    npm run build
+    ```
+
+5. Start the local web server, which runs in Node.js.
+
+    * If you'll be using Excel for Windows to test your custom functions, run the following command to start the local web server, launch Excel, and sideload the add-in:
+
+        ```
+         npm run start
+        ```
+        After running this command, your command prompt will show details about what has been done, another npm window will open showing the details of the build, and Excel will start with your add-in loaded. If you add-in does not load, check that you have completed step 3 properly.
+
+    * If you'll be using Excel Online to test your custom functions, run the following command to start the local web server:
+
+        ```
+        npm run start-web
+        ```
+
+         After running this command, another window will open showing you the details of the build. To use your functions, open a new workbook in Office Online.
+
+## Try out a prebuilt custom function
+
+The custom functions project that you created by using the Yeoman generator contains some prebuilt custom functions, defined within the **src/customfunctions.js** file. The **manifest.xml** file in the root directory of the project specifies that all custom functions belong to the `CONTOSO` namespace.
+
+In your Excel workbook, try out the `ADD` custom function by completing the following steps in Excel:
+
+1. Within a cell, type **=CONTOSO**. Notice that the autocomplete menu shows the list of all functions in the `CONTOSO` namespace.
+
+2. Run the `CONTOSO.ADD` function, with numbers `10` and `200` as input parameters, by typing the value `=CONTOSO.ADD(10,200)` in the cell and pressing enter.
+
+The `ADD` custom function computes the sum of the two numbers that you specify as input parameters. Typing `=CONTOSO.ADD(10,200)` should produce the result **210** in the cell after you press enter.
+
+## Create a custom function that requests data from the web
```

Comment:
We should consider adding a tutorial step to teach the user how to make a "simple" custom function first.

Reply:
This PR is mainly refactoring the *file structure* of the existing tutorial; I'll defer this feedback to @mscharlock for future consideration.

Reply:
Thanks for the suggestion @AlexJerabek - we originally had this back in the day but the feedback I got was that having a "simple" one felt unnecessary/too reductive given you can look at some built-in simple ones. Happy to reconsider if you felt that from your own experience going through the tutorial it was necessary.